src/Compilers/CSharp/Test/Semantic/Semantics/UsingStatementTests.cs

```diff
@@ -1783,6 +1783,151 @@ await using (C1 c = new C1())
             CompileAndVerify(compilation, expectedOutput: "Dispose async; Dispose; ");
         }
 
+        [Fact]
+        public void UsingPatternExtensionMethodWithNullableValue()
+        {
+            var source = @"
+static class C1 
+{
+    internal static void Dispose(this int x)
+    {
+        System.Console.Write($""Dispose {x}; "");
+    }
+}
+ class C2
+{
+    static void Main()
+    {
+        using (int? i1 = 1)
+        {
+            // Dispose 1
+        }
+        
+        int? i2 = 2;
+        using (i2) 
+        {
+            // Dispose 2
+        }       
+        
+        using (int? i3 = null)
+        {
+            // no dispose, null
+        }
+        
+        int? i4 = null;
+        using (i4) 
+        {
+            // no dispose, null
+        }
+    }
+}";
+            var compilation = CreateCompilation(source, options: TestOptions.DebugExe).VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: "Dispose 1; Dispose 2; ");
+        }
+
+        [Fact]
+        public void UsingPatternExtensionMethodWithNullableValueTargetAndValueTarget()
+        {
+            var source = @"
+static class C2 
+{
+    internal static void Dispose(this int? c1)
+    {
+        System.Console.Write($""Dispose? {c1}; "");
+    }        
+    
+    internal static void Dispose(this int c1)
+    { 
+        System.Console.Write($""Dispose {c1}; "");
+    }
+}
+ class C3
+{
+    static void Main()
+    {
+        using (int? i1 = 1)
+        {
+            // Dispose 1
+        }
+        
+        int? i2 = 2;
+        using (i2)
+        {
+            // Dispose 2
+        }
+      
+        using (int i3 = 3)
+        {
+            // Dispose 3
+        }
+        
+        int i4 = 4;
+        using (i4)
+        {
+            // Dipose 4        
+        }
+    }
+}";
+            var compilation = CreateCompilation(source, options: TestOptions.DebugExe).VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: "Dispose 1; Dispose 2; Dispose 3; Dispose 4;");
```

Comment:
>Dispose 1; Dispose 2; Dispose 3; Dispose 4; [](start = 59, length = 43)

It's surprising we ignore `Dispose(this int?)`. If that's the expected behavior, please add a test that where there is only `Dispose(this T?)` so it's clear.

Reply:
Yes, this was an explicit decision made at LDM; we'll never call it with a null value, so we only target the non-nullable extension method (essentially equivalent to doing `x?.Dispose()`). This is also how GetPinnableAddress works.